llvm/test/Transforms/InstCombine/select-factorize.ll

```diff
@@ -303,7 +303,7 @@ define i1 @and_logic_and_logic_or_not_one_use(i1 %c, i1 %a, i1 %b) {
 ; CHECK-LABEL: @and_logic_and_logic_or_not_one_use(
 ; CHECK-NEXT:    [[AC:%.*]] = and i1 [[A:%.*]], [[C:%.*]]
 ; CHECK-NEXT:    [[BC:%.*]] = select i1 [[B:%.*]], i1 [[C]], i1 false
-; CHECK-NEXT:    [[OR:%.*]] = select i1 [[BC]], i1 true, i1 [[AC]]
+; CHECK-NEXT:    [[OR:%.*]] = select i1 [[B]], i1 [[C]], i1 [[AC]]
```

Comment:
I'd consider this a regression, as logical or is more canonical. Skip if the outer select is already a logical op?

Reply:
(Or maybe if the operand is already constant?)

Reply:
(Though it would be an improvement if the condition weren't multi-use, so maybe it's okay to just ignore this.)